File: php/kode program/Product.php
<?php
class Product {
    // atribut privat, hanya bisa diakses lewat getter dan setter
    private $id;     // id produk
    private $name;   // nama produk
    private $price;  // harga produk
    private $stock;  // stok produk
    private $image;  // path gambar produk

    // constructor untuk inisialisasi objek product
    public function __construct($id, $name, $price, $stock, $image) {
        $this->id = $id;
        $this->name = $name;
        $this->price = $price;
        $this->stock = $stock;
        $this->image = $image;
    }

    // getter untuk ambil nilai atribut
    public function getId() { return $this->id; }
    public function getName() { return $this->name; }
    public function getPrice() { return $this->price; }
    public function getStock() { return $this->stock; }
    public function getImage() { return $this->image; }

    // setter untuk ubah nilai atribut tertentu
    public function setName($name) { $this->name = $name; }
    public function setPrice($price) { $this->price = $price; }
    public function setStock($stock) { $this->stock = $stock; }
    public function setImage($image) { $this->image = $image; }
}
?>

File: cpp/kode program/Product.cpp
#include <iostream>
#include <string>
using namespace std;

class Product {
private:
    int id;
    string name;
    int price;
    int stock;
    string image;

public:
    // constructor
    Product() {
    }

    Product(int id, string name, int price, int stock, string image) {
        this->id = id;
        this->name = name;
        this->price = price;
        this->stock = stock;
        this->image = image;
    }

    // setter dan getter untuk id
    void setId(int id) {
        this->id = id;
    }
    int getId() {
        return this->id;
    }

    // setter dan getter untuk name
    void setName(string name) {
        this->name = name;
    }
    string getName() {
        return this->name;
    }

    // setter dan getter untuk price
    void setPrice(int price) {
        this->price = price;
    }
    int getPrice() {
        return this->price;
    }

    // setter dan getter untuk stock
    void setStock(int stock) {
        this->stock = stock;
    }
    int getStock() {
        return this->stock;
    }

    // setter dan getter untuk image
    void setImage(string image) {
        this->image = image;
    }
    string getImage() {
        return this->image;
    }
};